### PR TITLE
Address floc review comments

### LINKED ISSUE #6639: Firebase Segmentation SDK clean up and refactoring
Firebase Segmentation SDK (beta) is merged into master in https://github.com/firebase/firebase-ios-sdk/pull/6466. However there are still some un-addressed minor issues. Please address them.

### PR DESCRIPTION
**Pull request by morganchen12**
Fixes #6639.

#no-changelog since Segmentation isn't released yet

### CODE REVIEW

File: FirebaseCore/Sources/FIRLogger.m [hunk 1]
```diff
@@ -28,6 +28,7 @@
 FIRLoggerService kFIRLoggerMLKit = @"[Firebase/MLKit]";
 FIRLoggerService kFIRLoggerPerf = @"[Firebase/Performance]";
 FIRLoggerService kFIRLoggerRemoteConfig = @"[Firebase/RemoteConfig]";
```
Comment (paulb777) [inline FirebaseCore/Sources/FIRLogger.m:30]: Delete. See comment above. FirebaseCore should not know about its clients :)

File: FirebaseCore/Sources/Private/FIRLogger.h [hunk 2]
```diff
@@ -31,6 +31,7 @@ extern FIRLoggerService kFIRLoggerCore;
 extern FIRLoggerService kFIRLoggerMLKit;
 extern FIRLoggerService kFIRLoggerPerf;
 extern FIRLoggerService kFIRLoggerRemoteConfig;
+extern FIRLoggerService kFIRLoggerSegmentation;
```
Comment (paulb777) [inline FirebaseCore/Sources/Private/FIRLogger.h:34]: Delete

**Review by paulb777 — COMMENTED** [review]
Doesn't have to be in this PR, but would be nice to add GHA to run the unit tests and build the sample app.

File: FirebaseCore/Sources/FIRLogger.m (hunk 1 above)
Comment (morganchen12) [reply FirebaseCore/Sources/FIRLogger.m]: Done.

File: FirebaseCore/Sources/Private/FIRLogger.h (hunk 2 above)
Comment (morganchen12) [reply FirebaseCore/Sources/Private/FIRLogger.h]: Done.

**Comment by morganchen12** [comment]
Filed #6703 for GHA changes.

**Review by paulb777 — APPROVED** [review]
LGTM with format fixes.

**Pull request merged**